bazel/foreign_cc/BUILD

```diff
@@ -253,106 +253,6 @@ envoy_cmake(
     }),
 )
 
-envoy_cmake(
```

Comment:
perhaps a one line comment about why we don't need this would be helpful?

Also - I wonder - can we figure out a way to minimize this change? E.g. for windows this has `tags = ["skip_on_windows"]`, maybe we would benefit from something similar. In that case this change would fold to a single line + a comment on why we do that. This might make life easier when cutting a branch for a major new release (when we need to rebase/merge)? Just some thoughts.

Reply:
This - and other changes in this PR - got lost in the rebase. We did that in previous versions. The reason I did that this way back in 2.1 or 2.0? Lack of knowledge in Bazel. I'd love to simplify things where it's possible. Let's do this in a follow up though.